### ISSUE TITLE
Incorrect Encoder Generation with Reserved keywords

### ISSUE THREAD

**Issue by BinaryHexer** [dillonkearns/graphqelm#55]
It seems graphqelm will generate incorrect encoders if the record type contains reserved keywords like `type`.

Generated Code (Incorrect):
```
{-| Type for the Input input object.
-}
type alias Input =
    { apiVersion : String, type_ : String, payload : String }


{-| Encode a Input into a value that can be used as an argument.
-}
encodeInput : Input -> Value
encodeInput input =
    Encode.maybeObject
        [ ( "apiVersion", Encode.string input.apiVersion |> Just ), ( "type_", Encode.string input.type_ |> Just ), ( "payload", Encode.string input.payload |> Just ) ]
```
Schema:
```
input Input {
  apiVersion: String!
  type: String!
  payload: String!
}

type Mutation {
  processInput(input: Input!): JobEntity!
}
```
In the above example, `graphqelm` should generate encoder as follows:
```
{-| Encode a Input into a value that can be used as an argument.
-}
encodeInput : Input -> Value
encodeInput input =
    Encode.maybeObject
        [ ( "apiVersion", Encode.string input.apiVersion |> Just ), ( "type", Encode.string input.type_ |> Just ), ( "payload", Encode.string input.payload |> Just ) ]
```
Notice the lack of `_` in the string `type`

**Comment by dillonkearns** [comment]
@BinaryHexer thank you so much for reporting the issue! This is now fixed with [npm version 3.1.12](https://github.com/dillonkearns/graphqelm/blob/master/CHANGELOG-NPM-PACKAGE.md#3112---2018-04-15) 👍

**Comment by BinaryHexer** [comment]
Thanks for the quick resolution :)
